core/src/main/java/co/aikar/commands/BaseCommand.java

```diff
@@ -64,69 +65,228 @@
+
+    final Map<Class<?>, String> contextFlags = Maps.newHashMap(); // TODO: Unknown use-case
+
+    /**
+     * Contains the manager which owns and handles this command.
+     */
+    CommandManager<?, ?, ?, ?, ?, ?> manager = null;
+
+    /**
+     * The command which owns this. This may be null if there are no owners.
+     */
+    @Nullable BaseCommand parentCommand = null;
+
+    /**
+     * A map of all registered commands under this.
+     * A registered command is any subcommand.
+     */
+    Map<String, RootCommand> registeredCommands = new HashMap<>();
+
+    /**
+     * The description of the command. This may be null if no description has been provided.
+     */
+    @Nullable String description = null;
+
+    /**
+     * The name of the command. This may be null if no name has been provided.
+     */
+    @Nullable String commandName = null;
+
+    /**
+     * The permission of the command. This may be null if no permission has been provided.
+     */
+    @Nullable String permission = null;
+
+    /**
+     * The conditions of the command. This may be null if no conditions has been provided.
+     */
+    @Nullable String conditions = null;
+
+    /**
+     * The last operative context data of this command. This may be null if this command hasn't been run yet.
+     */
+    @Nullable CommandOperationContext lastCommandOperationContext = null;
+
+    /**
+     * The precommand handler to be used. This may not be null if the command has been registered correctly and a
+     * handler has been created in the implementation.
+     */
     @Nullable private MethodHandle preCommandHandler = null;
+
+    /**
+     * The precommand handler to be used if {@link #preCommandHandler} is null. Everything should be the same
+     * except for performance, which would be slower with this.
+     */
     private Method preCommandReflectiveHandler = null;
 
+    /**
+     * The handler of all uncaught exceptions thrown by the user's command implementation.
+     */
+    private ExceptionHandler exceptionHandler = null;
+
+    /**
+     * The name of the parenting subcommand.
+     */
+    private String parentSubcommand;
+
     @SuppressWarnings("WeakerAccess")
     private String execLabel;
     @SuppressWarnings("WeakerAccess")
     private String execSubcommand;
     @SuppressWarnings("WeakerAccess")
     private String[] origArgs;
-    CommandManager<?, ?, ?, ?, ?, ?> manager = null;
-    BaseCommand parentCommand;
-    Map<String, RootCommand> registeredCommands = new HashMap<>();
-    String description;
-    String commandName;
-    String permission;
-    String conditions;
 
-    private ExceptionHandler exceptionHandler = null;
-    CommandOperationContext lastCommandOperationContext;
-    private String parentSubcommand;
+    /**
+     * Creates a new instance of {@link BaseCommand} without any settings applied.
+     * You will have to manually write to {@link #commandName}.
```

Comment:
This isnt accurate. it will generate a command name automatically for you on register